src/plugins/plugin.legend.js

```diff
@@ -48,17 +48,35 @@ defaults._set('global', {
 			generateLabels: function(chart) {
 				var data = chart.data;
 				return helpers.isArray(data.datasets) ? data.datasets.map(function(dataset, i) {
+					var meta = chart.getDatasetMeta(i);
+					var controller = meta.controller;
+					var legendOpts = chart.options.legend;
+					var usePointStyle = legendOpts && legendOpts.labels && legendOpts.labels.usePointStyle;
+					var model;
```

Comment:
Could be wrong but does it make sense to use the already computed `_model` instead?

```javascript
var element = meta.dataset || (meta.data && meta.data[0]);
var model = element ? element._model : {};
```

Reply:
Same as above. `_model` cannot be used if there is no data in the dataset. Maybe we can use `_model` for lines only.

Reply:
`_resolve*Options` are specific and internal to our built-in controllers and should not be expected for external implementations, so I think this change will break some integrations.